src/main/java/uk/gov/hmcts/reform/roleassignment/controller/endpoints/GetAssignmentController.java

```diff
@@ -72,7 +72,9 @@ public GetAssignmentController(RetrieveRoleAssignmentOrchestrator retrieveRoleAs
         long etag = retrieveRoleAssignmentService.retrieveETag(actorId);
         String weakEtag = "W/\"" + etag + "\"";
         HttpHeaders responseHeaders = new HttpHeaders();
+        String corId = responseEntity.getHeaders().getFirst("x-correlation-id");
         responseHeaders.setETag(weakEtag);
+        responseHeaders.add(Constants.CORRELATION_ID_HEADER_NAME, corId);
         return ResponseEntity
             .status(HttpStatus.OK)
             .headers(responseHeaders)
```

Comment:
Had to do it like this because whenever I tried to set the Etag after setting the correlationId header, it would give a strange error.